src/txdb.cpp

```diff
@@ -70,10 +71,17 @@ struct CoinEntry {
 
 } // namespace
 
-CCoinsViewDB::CCoinsViewDB(fs::path ldb_path, size_t nCacheSize, bool fMemory, bool fWipe) :
-    m_db(std::make_unique<CDBWrapper>(ldb_path, nCacheSize, fMemory, fWipe, true)),
-    m_ldb_path(ldb_path),
-    m_is_memory(fMemory) { }
+CCoinsViewDB::CCoinsViewDB(fs::path ldb_path, size_t nCacheSize, bool fMemory, bool fWipe)
+    : m_db{new CDBWrapper{{
+          .db_path = ldb_path,
+          .cache_size = nCacheSize,
+          .in_memory = fMemory,
+          .wipe_existing = fWipe,
+          .obfuscate_data = true,
+          .do_compact = gArgs.GetBoolArg("-forcecompactdb", false),
```

Comment:
In commit "Use CDBWrapper::Options ctor for non-test callers" (c5566d4899d75d9030901092ecf82c97e277d6c2)

Moving `gArgs.GetBoolArg("-forcecompactdb", false)` call out of cdbwrapper code into txdb code seems like a step backwards or maybe sideways, since txdb code shouldn't be using ArgsManager either